### PR TITLE
How to guide for pull image from ECR using OIDC

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description

* Add how-to for configuring a job to pull and image from AWS ECR
* Update private-images page to link to this how to guide and present this method as the best-practice
* Add to the sidebar
* link to new page from main OIDC page

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Review by sagar-connect — APPROVED** [review]

File: jekyll/_cci2/private-images.adoc [hunk 1]
```diff
@@ -120,17 +122,30 @@ jobs:
...
+Some advantages to using OIDC over using AWS credentials to pull images from ECR:
+
+* **Improved security**: By using OIDC authentication, storing AWS credentials directly in your CircleCI configuration or environment variables is avoided, reducing the risk of exposure.
+
+* **Simplified credential management**: OIDC allows CircleCI to automatically manage the authentication process, eliminating the need to manually manage and rotate AWS credentials.
+
+* **Fine-grained access control**: By associating an IAM role with the OIDC authentication, exact permissions granted to CircleCI for pulling ECR images can be controlled, ensuring a least-privilege approach.
 
-You can start using private images from ECR in one of two ways:
+Alternatively, you can start use one of the following methods:
```
Comment (davidmdm) [inline jekyll/_cci2/private-images.adoc:145]: start *using*

**Review by davidmdm — APPROVED** [review]

File: jekyll/_cci2/pull-an-image-from-aws-ecr-with-oidc.adoc [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+= Pull an image from AWS ECR with OIDC
+:page-layout: classic-docs
+:page-liquid:
+:page-description: Learn how to pull an image from AWS ECR using OIDC authentication
+:icons: font
+:toc: macro
+:toc-title:
+
+Follow this how-to guide to configure a `docker` job to pull an image from AWS ECR using OIDC authentication.
+
```
Comment (kbatuigas) [inline jekyll/_cci2/pull-an-image-from-aws-ecr-with-oidc.adoc:15]:
```suggestion

[#introduction]
```

File: jekyll/_cci2/pull-an-image-from-aws-ecr-with-oidc.adoc [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+For alternative methods, see the xref:private-images#aws-ecr[the AWS ECR section] of the Docker authenticated pulls page.
+
+Some advantages to using OIDC over using AWS credentials to pull images from ECR:
+
+* **Improved security**: By using OIDC authentication, storing AWS credentials directly in your CircleCI configuration or environment variables is avoided, reducing the risk of exposure.
+
+* **Simplified credential management**: OIDC allows CircleCI to automatically manage the authentication process, eliminating the need to manually manage and rotate AWS credentials.
+
+* **Fine-grained access control**: By associating an IAM role with the OIDC authentication, exact permissions granted to CircleCI for pulling ECR images can be controlled, ensuring a least-privilege approach.
+
```
Comment (kbatuigas) [inline jekyll/_cci2/pull-an-image-from-aws-ecr-with-oidc.adoc:31]:
```suggestion

[#prerequisites]
```

File: jekyll/_cci2/pull-an-image-from-aws-ecr-with-oidc.adoc [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+== Prerequisites
+
+* A xref:first-steps#[CircleCI account] integrated with a supported VCS.
+* A project `.circleci/config.yml` file containing a job that uses the xref:using-docker#[Docker execution environment], for which you want to use a Docker image stored in AWS ECR.
+* Set up CircleCI as a trusted identity provider in AWS. Follow the xref:openid-connect-tokens#set-up-aws[Set up AWS instructions].
+
```
Comment (kbatuigas) [inline jekyll/_cci2/pull-an-image-from-aws-ecr-with-oidc.adoc:37]:
```suggestion

[#set-up-iam-role]
```

**Review by kbatuigas — CHANGES_REQUESTED** [review]
Just minor changes

**Review by kbatuigas — APPROVED** [review]

**Pull request merged**